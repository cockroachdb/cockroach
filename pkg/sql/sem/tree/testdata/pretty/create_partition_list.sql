CREATE TABLE students_by_list (
    id SERIAL,
    name STRING,
    email STRING,
    country STRING,
    expected_graduation_date DATE,   
    PRIMARY KEY (country, id))
    PARTITION BY LIST (country)
      (PARTITION north_america VALUES IN ('CA','US'),
      PARTITION australia VALUES IN ('AU','NZ'),
      PARTITION DEFAULT VALUES IN (default))
