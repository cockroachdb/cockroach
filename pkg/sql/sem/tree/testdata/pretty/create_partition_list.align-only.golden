// Code generated by TestPretty. DO NOT EDIT.
// GENERATED FILE DO NOT EDIT
1:
-
CREATE TABLE students_by_list (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (
		country,
		id
	)
)
	PARTITION BY LIST (
		country
	)
		(
			PARTITION north_america
				VALUES IN (
					'CA',
					'US'
				),
			PARTITION australia
				VALUES IN (
					'AU',
					'NZ'
				),
			PARTITION default
				VALUES IN (
					DEFAULT
				)
		)

23:
-----------------------
CREATE TABLE students_by_list (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (
		country,
		id
	)
)
	PARTITION BY LIST (
		country
	) (
		PARTITION north_america
			VALUES IN (
				'CA',
				'US'
			),
		PARTITION australia
			VALUES IN (
				'AU',
				'NZ'
			),
		PARTITION default
			VALUES IN (
				DEFAULT
			)
	  )

29:
-----------------------------
CREATE TABLE students_by_list (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (country, id)
)
	PARTITION BY LIST (
		country
	) (
		PARTITION north_america
			VALUES IN (
				'CA',
				'US'
			),
		PARTITION australia
			VALUES IN (
				'AU',
				'NZ'
			),
		PARTITION default
			VALUES IN (
				DEFAULT
			)
	  )

31:
-------------------------------
CREATE TABLE students_by_list (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america
									VALUES IN (
										'CA',
										'US'
									),
								PARTITION australia
									VALUES IN (
										'AU',
										'NZ'
									),
								PARTITION default
									VALUES IN (
										DEFAULT
									)
                              )

37:
-------------------------------------
CREATE TABLE students_by_list (
	id                       SERIAL8,
	name                     STRING,
	email                    STRING,
	country                  STRING,
	expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america
									VALUES IN (
										'CA',
										'US'
									),
								PARTITION australia
									VALUES IN (
										'AU',
										'NZ'
									),
								PARTITION default
									VALUES IN (
										DEFAULT
									)
                              )

55:
-------------------------------------------------------
CREATE TABLE students_by_list (
	id                       SERIAL8,
	name                     STRING,
	email                    STRING,
	country                  STRING,
	expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america
									VALUES IN (
										'CA',
										'US'
									),
								PARTITION australia
									VALUES IN (
										'AU',
										'NZ'
									),
								PARTITION default
									VALUES IN (DEFAULT)
                              )

59:
-----------------------------------------------------------
CREATE TABLE students_by_list (
	id                       SERIAL8,
	name                     STRING,
	email                    STRING,
	country                  STRING,
	expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america
									VALUES IN ('CA', 'US'),
								PARTITION australia
									VALUES IN ('AU', 'NZ'),
								PARTITION default
									VALUES IN (DEFAULT)
                              )

61:
-------------------------------------------------------------
CREATE TABLE students_by_list (
	id                       SERIAL8,
	name                     STRING,
	email                    STRING,
	country                  STRING,
	expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america
									VALUES IN ('CA', 'US'),
								PARTITION australia
									VALUES IN ('AU', 'NZ'),
								PARTITION default VALUES IN (
													DEFAULT
								                  )
                              )

63:
---------------------------------------------------------------
CREATE TABLE students_by_list (
	id                       SERIAL8,
	name                     STRING,
	email                    STRING,
	country                  STRING,
	expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america
									VALUES IN ('CA', 'US'),
								PARTITION australia VALUES IN (
														'AU',
														'NZ'
								                    ),
								PARTITION default VALUES IN (
													DEFAULT
								                  )
                              )

67:
-------------------------------------------------------------------
CREATE TABLE students_by_list (
	id                       SERIAL8,
	name                     STRING,
	email                    STRING,
	country                  STRING,
	expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america VALUES IN (
															'CA',
															'US'
								                        ),
								PARTITION australia VALUES IN (
														'AU',
														'NZ'
								                    ),
								PARTITION default VALUES IN (
													DEFAULT
								                  )
                              )

69:
---------------------------------------------------------------------
CREATE TABLE students_by_list (
	id                       SERIAL8,
	name                     STRING,
	email                    STRING,
	country                  STRING,
	expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america VALUES IN (
															'CA',
															'US'
								                        ),
								PARTITION australia VALUES IN (
														'AU',
														'NZ'
								                    ),
								PARTITION default VALUES IN (DEFAULT)
                              )

75:
---------------------------------------------------------------------------
CREATE TABLE students_by_list (
	id                       SERIAL8,
	name                     STRING,
	email                    STRING,
	country                  STRING,
	expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america VALUES IN (
															'CA',
															'US'
								                        ),
								PARTITION australia VALUES IN ('AU', 'NZ'),
								PARTITION default VALUES IN (DEFAULT)
                              )

79:
-------------------------------------------------------------------------------
CREATE TABLE students_by_list (
	id                       SERIAL8,
	name                     STRING,
	email                    STRING,
	country                  STRING,
	expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america VALUES IN ('CA', 'US'),
								PARTITION australia VALUES IN ('AU', 'NZ'),
								PARTITION default VALUES IN (DEFAULT)
                              )

89:
-----------------------------------------------------------------------------------------
CREATE TABLE students_by_list (
	id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (
								PARTITION north_america VALUES IN ('CA', 'US'),
								PARTITION australia VALUES IN ('AU', 'NZ'),
								PARTITION default VALUES IN (DEFAULT)
                              )

161:
-----------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (
	id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE,
	PRIMARY KEY (country, id)
) PARTITION BY LIST (country) (PARTITION north_america VALUES IN ('CA', 'US'), PARTITION australia VALUES IN ('AU', 'NZ'), PARTITION default VALUES IN (DEFAULT))

163:
-------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (
																																					country
                                                                                                                                                ) (
																																					PARTITION north_america
																																						VALUES IN (
																																							'CA',
																																							'US'
																																						),
																																					PARTITION australia
																																						VALUES IN (
																																							'AU',
																																							'NZ'
																																						),
																																					PARTITION default
																																						VALUES IN (
																																							DEFAULT
																																						)
                                                                                                                                                  )

171:
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (
																																					country
                                                                                                                                                ) (
																																					PARTITION north_america
																																						VALUES IN (
																																							'CA',
																																							'US'
																																						),
																																					PARTITION australia
																																						VALUES IN (
																																							'AU',
																																							'NZ'
																																						),
																																					PARTITION default
																																						VALUES IN (DEFAULT)
                                                                                                                                                  )

173:
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (country) (
																																												PARTITION north_america
																																													VALUES IN (
																																														'CA',
																																														'US'
																																													),
																																												PARTITION australia
																																													VALUES IN (
																																														'AU',
																																														'NZ'
																																													),
																																												PARTITION default
																																													VALUES IN (
																																														DEFAULT
																																													)
                                                                                                                                                                            )

199:
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (country) (
																																												PARTITION north_america
																																													VALUES IN (
																																														'CA',
																																														'US'
																																													),
																																												PARTITION australia
																																													VALUES IN (
																																														'AU',
																																														'NZ'
																																													),
																																												PARTITION default
																																													VALUES IN (DEFAULT)
                                                                                                                                                                            )

203:
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (country) (
																																												PARTITION north_america
																																													VALUES IN ('CA', 'US'),
																																												PARTITION australia
																																													VALUES IN ('AU', 'NZ'),
																																												PARTITION default
																																													VALUES IN (DEFAULT)
                                                                                                                                                                            )

205:
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (country) (
																																												PARTITION north_america
																																													VALUES IN ('CA', 'US'),
																																												PARTITION australia
																																													VALUES IN ('AU', 'NZ'),
																																												PARTITION default VALUES IN (
																																																	DEFAULT
																																												                  )
                                                                                                                                                                            )

207:
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (country) (
																																												PARTITION north_america
																																													VALUES IN ('CA', 'US'),
																																												PARTITION australia VALUES IN (
																																																		'AU',
																																																		'NZ'
																																												                    ),
																																												PARTITION default VALUES IN (
																																																	DEFAULT
																																												                  )
                                                                                                                                                                            )

211:
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (country) (
																																												PARTITION north_america VALUES IN (
																																																			'CA',
																																																			'US'
																																												                        ),
																																												PARTITION australia VALUES IN (
																																																		'AU',
																																																		'NZ'
																																												                    ),
																																												PARTITION default VALUES IN (
																																																	DEFAULT
																																												                  )
                                                                                                                                                                            )

213:
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (country) (
																																												PARTITION north_america VALUES IN (
																																																			'CA',
																																																			'US'
																																												                        ),
																																												PARTITION australia VALUES IN (
																																																		'AU',
																																																		'NZ'
																																												                    ),
																																												PARTITION default VALUES IN (DEFAULT)
                                                                                                                                                                            )

219:
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (country) (
																																												PARTITION north_america VALUES IN (
																																																			'CA',
																																																			'US'
																																												                        ),
																																												PARTITION australia VALUES IN ('AU', 'NZ'),
																																												PARTITION default VALUES IN (DEFAULT)
                                                                                                                                                                            )

223:
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (country) (
																																												PARTITION north_america VALUES IN ('CA', 'US'),
																																												PARTITION australia VALUES IN ('AU', 'NZ'),
																																												PARTITION default VALUES IN (DEFAULT)
                                                                                                                                                                            )

303:
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students_by_list (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, id)) PARTITION BY LIST (country) (PARTITION north_america VALUES IN ('CA', 'US'), PARTITION australia VALUES IN ('AU', 'NZ'), PARTITION default VALUES IN (DEFAULT))


